### DISCUSSION TITLE
aps-environment in iOS let app crash with certificate error

### DISCUSSION THREAD

**Discussion by Marcophono2** [airsdk/Adobe-Runtime-Support#1554] [Q&A] [answered]
System: Flash Builder 4.7 on Windows 10, latest AIR SDK 33.1.1.713
After a year or so I wanted to update my app but had to renew my Apple certificates (distribution, development including mobilie provision file). I created the certs on my brandnew M1 Mac mini and exported them to .p12. (not sure if the new Mac is being a factor here, so I prefer to tell this)
When I then try to launch my app in development mode on my iPhone XR with this settings

```
		<Entitlements>
            <![CDATA[
		<!-- DEVELOPMENT -->
	
		<key>get-task-allow</key>
		<true/>
		<key>aps-environment</key>
		<string>development</string>
		
		<key>application-identifier</key>
		<string>xxxxxxxxxxxxx.marc.waesche.org.marcophono</string>
		<key>keychain-access-groups</key>
		<array>
			<string>xxxxxxxxxxxx.marc.waesche.org.marcophono</string>
		</array>            
            ]]>
		</Entitlements>
```

Flash Builder returns this error:

```
Error occurred while installing the application:
WARNING: could not locate iTunesMetadata.plist in archive!
WARNING: could not locate Payload/Marcophono.app/SC_Info/Marcophono.sinf in archive!
ERROR: Install failed. Got error "ApplicationVerificationFailed" with code 0xe8008015: Failed to verify code signature of /var/installd/Library/Caches/com.apple.mobile.installd.staging/temp.1hVmvb/extracted/Payload/Marcophono.app : 0xe8008015 (A valid provisioning profile for this executable was not found.)
```

But if I just remove the part

```
		<key>get-task-allow</key>
		<true/>
		<key>aps-environment</key>
		<string>development</string>
```

the app launches without any problem. 
Sure, that is a perfect workaround but I would like to know the reason behind it. Otherwise there could come up a cert issue after publishing the app to the App Store. I have the same problem if I export the app with an adhoc distribution cert and try to launch it on my iPhone. With the production instead of the development setting in the XML file of course. 

Thank you and best regards
Marc

**Comment by marchbold** [comment, marked as answer]
This sounds like an issue with the provisioning profile you are using. You must enable push in your profile in the developer console, then download and install the updated profile. 

Without this in your profile adding entitlements not found in the profile will cause a verification failure.

**Reply by Marcophono2** [reply]
Yeah, that was it! I absolutely overlooked that I had to check a button next to "push" when I created the "identity". It seems that I recreated the identities may be as result of Apple's new certificate security rules. To be honest I cannot remember why and when I recreated the identities. Hmmm, or it had to do with an outrun Push certificate.... Anyway, it works now. Thans to Michael! :)

**Reply by marchbold** [reply]
Glad that helped
